### DISCUSSION TITLE
Using existing values in model definition functions

### DISCUSSION THREAD

**Discussion by mauriceoc** [mswjs/data#198] [Q&A] [answered]
Hey folks,

Been using the data library for a couple of things and it's going great. Just encountered an issue though -

I have a model definition using `faker` like this:

```javascript
user: {
        id: primaryKey(faker.datatype.uuid),
        firstName: faker.name.firstName,
        lastName: faker.name.lastName,
        email: faker.internet.exampleEmail
    }
```

This works fine, but I'd like the user's email to be generated with correct `firstName` and `lastName`, so something like this:

```javascript
user: {
        id: primaryKey(faker.datatype.uuid),
        firstName: faker.name.firstName,
        lastName: faker.name.lastName,
        email: () => faker.internet.exampleEmail(??, ??)
    }
```

Could be my javascript needs polishing but not sure how I can get access to the existing `firstName` and `lastName` values within the `email` function! Any thoughts?

**Comment by mauriceoc** [comment]
Can see this question is up for discussion in [this issue](https://github.com/mswjs/data/issues/95), so will follow that.

**Comment by kettanaito** [comment, marked as answer]
## Update

Since v1.0, Data no longer provides any object modeling capabilities. You can use any Standard Schema-compliant library (Zod, ArkType, etc) to describe a schema where some properties are derived from other properties. 

Here's an example of deriving the `email` field from the `firstsName` and `lastName` fields:

https://github.com/mswjs/data/blob/471a1f6154103a50eedd61f938fb2c22cd9a3af7/tests/create-derived-property.test.ts#L5-L16

You should do this on the schema level. This means that you should consult your schema library of choice on the exact APIs to achieve such derived values.
